pulsar-function-go/pf/instance.go

```diff
@@ -0,0 +1,289 @@
+
+	properties := getProperties(getDefaultSubscriptionName(
+		funcDetails.Tenant,
+		funcDetails.Namespace,
+		funcDetails.Name), gi.context.InstanceConf.InstanceID)
+
+	channel := make(chan pulsar.ConsumerMessage)
+
+	for topic, consumerConf := range funcDetails.Source.InputSpecs {
+		log.Debugf("Setting up consumer for topic: %s with subscription name: %s", topic, subscriptionName)
+		if consumerConf.ReceiverQueueSize != nil {
+			if consumerConf.IsRegexPattern {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:     topic,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              consumerType,
+					MessageChannel:    channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			} else {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:             topic,
+					SubscriptionName:  subscriptionName,
+					Properties:        properties,
+					Type:              consumerType,
+					ReceiverQueueSize: int(consumerConf.ReceiverQueueSize.Value),
+					MessageChannel:    channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+
+				gi.consumers = append(gi.consumers, consumer)
+			}
+		} else {
+			if consumerConf.IsRegexPattern {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					TopicsPattern:    topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             consumerType,
+					MessageChannel:   channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			} else {
+				consumer, err := gi.client.Subscribe(pulsar.ConsumerOptions{
+					Topic:            topic,
+					SubscriptionName: subscriptionName,
+					Properties:       properties,
+					Type:             consumerType,
+					MessageChannel:   channel,
+				})
+				if err != nil {
+					log.Errorf("create consumer error:%s", err.Error())
+				}
+				gi.consumers = append(gi.consumers, consumer)
+			}
+		}
+		gi.context.InputTopics = append(gi.context.InputTopics, topic)
+
+	CLOSE:
+		for {
+			select {
+			case cm := <-channel:
+				gi.msg = append(gi.msg, cm.Message)
+			case <-time.After(time.Millisecond * 500):
+				close(channel)
+				break CLOSE
+			}
+		}
+	}
+
+	gi.actualExecution(function)
+	gi.close()
+}
+
+func (gi *GoInstance) actualExecution(function Function) {
+	gi.function = function
+	output, err := gi.handlerMsg()
+	err = gi.processResult(output)
+	if err != nil {
+		log.Errorf("process output result err:%v", err)
+		return
+	}
+}
+
+func (gi *GoInstance) processResult(output []byte) error {
+	gi.context = NewFuncContext()
```

Comment:
context is already created when `gi.setupContext()` at `startFunction`

Reply:
done